### PR TITLE
Python3 compatible

### PR DESCRIPTION
**Pull request by DSil**
Used 2to3 and fixed a small "bug" that caused Python3 version not to convert to int, in smooth.py.

Corrected README.md since Track.from_gpx just needs the file path (as a string) as an argument.

### CODE REVIEW

File: tracktotrip/learn_trip.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def learn_trip(current, current_id, canonical_trips, insert_canonical, update_ca
 
     if len(canonical_trips) == 0:
         current.simplify(eps, 0, 0, topology_only=True)
-        print("inserting trip %d" % len(current.points))
+        print(("inserting trip %d" % len(current.points)))
```
Comment (ruipgil) [inline tracktotrip/learn_trip.py]: Why are you changing prints so that they have double parenthesis?

File: tracktotrip/location.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 FS_CACHE = {}
 
 def from_cache(cache, point, threshold):
-    for entry in cache.keys():
+    for entry in list(cache.keys()):
```
Comment (ruipgil) [inline tracktotrip/location.py]: There's no need to convert a set into a list.

File: tracktotrip/location.py [hunk 3]
```diff
@@ -216,7 +216,7 @@ def infer_location(
         queried_locations = location_query(point, max_distance)
         for (label, centroid, _) in queried_locations:
             locations.append({
-                'label': unicode(label, 'utf-8'),
+                'label': str(label, 'utf-8'),
```
Comment (ruipgil) [inline tracktotrip/location.py:219]: A label might have a unicode character

File: tracktotrip/track.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 from os.path import basename
 from datetime import timedelta
 
-import gpxpy
+from . import gpxpy
```
Comment (ruipgil) [inline tracktotrip/track.py]: `gpxpy` is a dependency, it should not be imported like this.

**Review by ruipgil — CHANGES_REQUESTED** [review]

File: tracktotrip/learn_trip.py (hunk 1 above)
Comment (DSil) [reply tracktotrip/learn_trip.py]: Automatically done by 2to3, I don't think it makes any difference, but can be done to have just one pair of parenthesis.

File: tracktotrip/location.py (hunk 2 above)
Comment (DSil) [reply tracktotrip/location.py]: Automatically done by 2to3, I don't think it makes any difference, but can be done to have no conversion.

File: tracktotrip/location.py (hunk 3 above)
Comment (DSil) [reply tracktotrip/location.py:219]: Literal strings are unicode by default in Python3. Unicode of Python2 is equivalent to str in Python3

File: tracktotrip/track.py (hunk 4 above)
Comment (DSil) [reply tracktotrip/track.py]: I believe it was introduced by mistake by me. Will fix to the original form.

File: tracktotrip/location.py (hunk 3 above)
Comment (ruipgil) [reply tracktotrip/location.py:219]: yes, but that will break compatibility with python 2.x

File: tracktotrip/transportation_mode.py [hunk 5]
```diff
@@ -192,7 +192,7 @@ def group_modes(modes):
 
 def classify(clf, points, min_time, from_index = None, to_index = None):
     features = extract_features_2(points)
-    print(len(points), features)
+    print((len(points), features))
```
Comment (ruipgil) [inline tracktotrip/transportation_mode.py:195]: This still has two parenthesis.

**Review by ruipgil — CHANGES_REQUESTED** [review]

File: tracktotrip/transportation_mode.py (hunk 5 above)
Comment (stephanobryan) [reply tracktotrip/transportation_mode.py:195]: Is it compatible with Python 3 now?

**Comment by DSil** [comment]
Closing this due to inactivity and lack of time to move this on.

**Pull request closed without merging**